AWSIoTPythonSDK/core/greengrass/discovery/providers.py

```diff
@@ -368,7 +368,7 @@ def _receive_until(self, ssl_sock, criteria_function, extra_data=None):
         number_bytes_read = 0
         while True:  # Python does not have do-while
             try:
-                response.append(self._convert_to_int_py3(ssl_sock.read(1)))
+                response.extend(self._convert_to_int_py3(ssl_sock.read(1)))
```

Comment:
It looks like there is an issue with this line and integers, as the unit tests fail mentioning that integers are not iterable. ([error line 401](https://github.com/aws/aws-iot-device-sdk-python/runs/5238081505?check_suite_focus=true)). I'm not sure where it's iterating though.